src/quemb/molbe/graphfrag.py

```diff
@@ -0,0 +1,608 @@
+    if isinstance(n_BE, str):
+        fragment_type_order = int(re.findall(r"\d+", str(n_BE))[0])
+    else:
+        fragment_type_order = n_BE
+    if cutoff == 0.0 and fragment_type_order <= 3:
+        cutoff = 4.5
+    elif cutoff == 0.0:
+        cutoff = 4.5 * fragment_type_order
+
+    natm: int = mol.natm
+
+    # Used to be `fsites`
+    AO_per_frag: list[Sequence[int]] = list(tuple())
+
+    # Similar to `AO_per_frag`: Ao indices per fragment per atom.
+    fsites_by_atom: list[Sequence[Sequence[int]]] = list(tuple(tuple()))
+
+    # Used to be `edge_sites`
+    AO_per_edge_per_frag: list[Sequence[Sequence[int]]] = list(tuple(tuple()))
+
+    # Center site AO indices per fragment.
+    center: list[Sequence[int]] = list(tuple())
+
+    # Used to be `centerf_idx`
+    relAO_per_origin_per_frag: list[Sequence[int]] = list(tuple())
+
+    # Used to be `ebe_weight`
+    weight_and_relAO_per_center_per_frag: list[Sequence] = list(tuple())
+    sites: list[Sequence] = list(tuple())
+    dnames: list[str] = list()
+    motifs_per_frag: list[Sequence[int]] = list()
+    origin_per_frag: list[Sequence[int]] = list()
+
+    # List of edge atom indices per fragment
+    edge_atoms: list[Sequence[int]] = list()
+
+    # The full molecular adjacency matrix
+    adjacency_mat: np.ndarray = np.zeros((natm, natm), np.float64)
+
+    # The molecular adjacency graph, constructed from `adjacency_mat`
+    adjacency_graph: nx.Graph = nx.Graph()
+
+    # List of graph edges to added to `adjacency_graph`
+    edge_list: list[Sequence] = list()
+
+    # Center atom indices that have been added to each fragment via
+    # `remove_nonnunique_frags`
+    add_center_atom: list[Sequence] = list()
+
+    # Dictionary mapping atom index (adx) to relevant atomic info.
+    # Each entry to `adx_map` is a node in the adjacency graph.
+    adx_map: dict = {
+        adx: {
+            "bas": bas,
+            "label": mol.atom_symbol(adx),
+            "coord": mol.atom_coord(adx),
+            "shortest_paths": dict(),
+            "attached_hydrogens": list(),
+        }
+        for adx, bas in enumerate(mol.aoslice_by_atom())
+    }
+
+    adjacency_graph.add_nodes_from(adx_map)
+
+    _core_offset = 0
+    for adx, map in adx_map.items():
+        start_ = map["bas"][2]
+        stop_ = map["bas"][3]
+        if frozen_core:
+            # When frozen_core=True, offset all AO indices by the
+            # number of core orbitals per atom.
+            _, _, core_list = get_core(mol)
+            start_ -= _core_offset
+            ncore_ = int(core_list[adx])
+            stop_ -= _core_offset + ncore_
+            _core_offset += ncore_
+            sites.append(tuple([i for i in range(start_, stop_)]))
+        else:
+            # When frozen_core=False, each 'site' is all AO indices
+            # per atom.
+            sites.append(tuple([i for i in range(start_, stop_)]))
+
+    if connectivity.lower() in ["euclidean_distance", "euclidean"]:
```

Comment:
```suggestion
    if connectivity == "euclidean":
```
or
```suggestion
```
since `"euclidean"` is the only option for now